From c86e48a7c5a6db0b6144aaeb1c3815837c3133d3 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Fri, 20 Oct 2017 19:01:58 +0800
Subject: [PATCH 2718/5242] MLK-16706-2 video: fbdev: mipi_dsi_northwest: add
 'phy_ref_clkfreq' field

commit  3fbcf814d810221ad82aa15d3ae327a6d97a1808 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add 'phy_ref_clkfreq' field to 'struct mipi_dsi_info'
to save the reference clock frequency configed in dtb
file for mipi phy pll.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.h           |    1 +
 drivers/video/fbdev/mxc/mipi_dsi_northwest.c |    1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.h b/drivers/video/fbdev/mxc/mipi_dsi.h
index b8d00b3..6219024 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.h
+++ b/drivers/video/fbdev/mxc/mipi_dsi.h
@@ -88,6 +88,7 @@ struct mipi_dsi_info {
 	int				disp_id;
 	char				*lcd_panel;
 	int				irq;
+	uint32_t			phy_ref_clkfreq;
 #ifdef CONFIG_FB_IMX64
 	struct clk			*core_clk;
 	struct clk			*phy_ref_clk;
diff --git a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
index d4dbff7..e7080c4 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi_northwest.c
@@ -985,6 +985,7 @@ static int dsi_clks_init(struct mipi_dsi_info *minfo)
 		dev_err(&pdev->dev, "invalid phy reference clock rate\n");
 		return -EINVAL;
 	}
+	minfo->phy_ref_clkfreq = phy_ref_clkfreq;
 
 	ret = clk_set_rate(minfo->phy_ref_clk, phy_ref_clkfreq);
 	if (ret < 0) {
-- 
1.7.9.5

